tools/pipeline/print-invalidations/Main.cpp

```diff
@@ -0,0 +1,88 @@
+/// \file Main.cpp
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <system_error>
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/DynamicLibrary.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "llvm/Support/raw_os_ostream.h"
+
+#include "revng/Model/LoadModelPass.h"
+#include "revng/Pipeline/AllRegistries.h"
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/CopyPipe.h"
+#include "revng/Pipeline/GenericLLVMPipe.h"
+#include "revng/Pipeline/Global.h"
+#include "revng/Pipeline/LLVMContainerFactory.h"
+#include "revng/Pipeline/LLVMGlobalKindBase.h"
+#include "revng/Pipeline/Loader.h"
+#include "revng/Pipeline/Step.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/ModelGlobal.h"
+#include "revng/Pipes/PipelineManager.h"
+#include "revng/Pipes/ToolCLOptions.h"
+#include "revng/Support/CommandLine.h"
+#include "revng/Support/InitRevng.h"
+#include "revng/TupleTree/TupleTree.h"
+
+using std::string;
+using namespace llvm;
+using namespace llvm::cl;
+using namespace pipeline;
+using namespace ::revng::pipes;
+using namespace revng;
+
+static cl::opt<std::string> DiffPath(cl::Positional,
+                                     cl::cat(MainCategory),
+                                     cl::desc("<model diff>"),
+                                     cl::init("-"),
+                                     cl::value_desc("model"));
+
+static ToolCLOptions BaseOptions(MainCategory);
+
+static ExitOnError AbortOnError;
+
+int main(int argc, const char *argv[]) {
+  using BinaryRef = TupleTreeGlobal<model::Binary>;
+  using DiffRef = TupleTreeDiff<model::Binary>;
+
+  revng::InitRevng X(argc, argv);
+
+  HideUnrelatedOptions(MainCategory);
+  ParseCommandLineOptions(argc, argv);
+
+  Registry::runAllInitializationRoutines();
+
+  auto Manager = AbortOnError(BaseOptions.makeManager());
+  const auto &Ctx = Manager.context();
+  auto OriginalModel = *AbortOnError(Ctx.getGlobal<BinaryRef>(ModelGlobalName));
+
+  const auto &Name = ModelGlobalName;
+  auto AfterModel = AbortOnError(Ctx.getGlobal<BinaryRef>(Name));
+
+  auto Diff = AbortOnError(deserializeFileOrSTDIN<DiffRef>(DiffPath));
+  revng::ErrorList EL;
+  Diff.apply(AfterModel->get(), EL);
+  cantFail(std::move(EL));
```

Comment:
What's going on here?

Reply:
`AbortOnError` was replaced with `cantFail`, after the introduction of `ErrorList`, and more specifically, after [this](https://github.com/revng/revng/commit/0b80801b2b75b3584b811148994a16dc4f184bf2) jackv's change, whose `apply` method was modified to accept a `ErrorList` too.

Reply:
Some concerns has [here](https://github.com/revng/revng/pull/209#discussion_r1021809669).

Reply:
this will disappear with no error lists